### DISCUSSION TITLE
Including the API key in a script

### DISCUSSION THREAD

**Discussion by ghost** [sanderkl/PSMeraki#27] [Q&A] [answered]
Hi there, How would I go about including the API key in a script so that I could interact with my tenancy non-interactively.

Thanks for the help!

Scott

**Comment by mattolan** [comment]
I would like to know how to do this also.  Did you ever find a solution?

**Reply by ghost** [reply]
Yes my friend, you:

top of your script
Set-MrkRestApiKey APIKEY

see https://www.powershellgallery.com/packages/PSMeraki/1.1.3/Content/Public%5CSet-MrkRestApiKey.ps1

**Comment by mattolan** [comment]
Thank you :-) 
I completely missed

**Comment by sanderkl** [comment, marked as answer]
As [Jellman01](https://github.com/Jellman01) suggested: Set-MrkRestApiKey APIKEY
But it gives you rest API key in clear text in a script, which is a problem for my current customer. 
An approach to workaround the plaintext password problem could be: 
```
get-credential | export-clixml -path C:\temp\Creds.xml
```
for the credentials, use a random user name, you don't need it. Set the api key as the password. 
Now you have a .xml file with encrypted password. 
to extract the password:
```
$creds = Import-Clixml -Path "C:\Temp\Cred.xml"
Set-MrkRestApiKey -key [System.Runtime.InteropServices.Marshal]::PtrToStringAuto([System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($creds.Password)) 
```
